server/src/test/java/org/elasticsearch/common/util/concurrent/EsExecutorsTests.java

```diff
@@ -655,6 +656,24 @@ public void testParseExecutorName() throws InterruptedException {
         }
     }
 
+    public void testScalingWithTaskTimeTracking() {
+        final int min = between(1, 3);
+        final int max = between(min + 1, 6);
+
+        ThreadPoolExecutor pool = EsExecutors.newScaling(
+            getClass().getName() + "/" + getTestName(),
+            min,
+            max,
+            between(1, 100),
+            randomTimeUnit(),
+            randomBoolean(),
+            EsExecutors.daemonThreadFactory("test"),
+            threadContext,
+            new EsExecutors.TaskTrackingConfig(true, 0.3)
+        );
```

Comment:
Would we also want a test to ensures that  `DO_NOT_TRACK` passed in here results in a `EsThreadPoolExecutor`?

Reply:
`EsExecutors.TaskTrackingConfig(true, 0.3));`
As an idea, this could be `EsExecutors.TaskTrackingConfig(randomBoolean(), 0.3));` as the creation of the `TaskExecutionTimeTrackingEsThreadPoolExecutor` should not depend on whether we track ongoing tasks.

Reply:
Good call. Done